packages/compass-query-bar/src/stores/query-bar-reducer.ts

```diff
@@ -440,6 +448,46 @@ const saveRecentQuery = (
   };
 };
 
+const updateFavoriteQuery = (
+  query: Omit<BaseQuery, 'maxTimeMS'>
+): QueryBarThunkAction<Promise<void>> => {
+  return async (
+    dispatch,
+    getState,
+    { favoriteQueryStorage, logger: { debug } }
+  ) => {
+    try {
+      const {
+        queryBar: { favoriteQueries },
+      } = getState();
+
+      const queryAttributes = getQueryAttributes(query);
+      // Ignore empty or default queries
+      if (isEmpty(queryAttributes)) {
+        return;
+      }
+
+      const existingFavoriteQuery = favoriteQueries.find((favoriteQuery) => {
+        return isQueryEqual(getQueryAttributes(favoriteQuery), queryAttributes);
```

Comment:
Since the favorite query is already being passed into this function and we're doing this same find there can we remove this check?
Then above we can pass `existingFavoriteQuery` to this function and make this function query argument type a bit stricter.